server/store/templateHelperFuncs.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package store
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost-plugin-user-survey/server/model"
+)
+
+func (s *SQLStore) genAddColumnIfNeeded(tableName, columnName, dataType, constraint string) (string, error) {
+	tableName = addPrefixIfNeeded(tableName, s.tablePrefix)
+	normalizedTableName := s.normalizeTablename(tableName)
+
+	switch s.dbType {
+	case model.DBTypeMySQL:
+		vars := map[string]string{
+			"schema":          s.schemaName,
+			"table_name":      tableName,
+			"norm_table_name": normalizedTableName,
+			"column_name":     columnName,
+			"data_type":       dataType,
+			"constraint":      constraint,
+		}
+
+		x := replaceVars(`
+			SET @stmt = (SELECT IF(
```

Comment:
Cannot we use query builders for this (and other stuff on this file)?

Reply:
Its unnecessarily complex to use query builder for this. Also, Gorm doesn't even support these stuff, we end up using raw queries anyways. We went through this idea in focalboard as well, where this code is copied from 😛